djweb/dj_comp_hist/views.py

```diff
@@ -196,101 +198,83 @@ def browse(request):
+
+    doc_objs = Document.objects
+
+    if search_params['title'] != '':
+        doc_objs = doc_objs.filter(Q(title__icontains=search_params['title']))
+
+    if search_params['text'] != '':
+        # allows quotation marks but only extracts the string in the middle
+        match = re.match('^[\'\"]?([a-zA-Z\d ]+)[\'\"]?$', search_params['text'])
+        if not match:
+            print(f"WARNING. Could not parse full text search string: {search_params['text']}.")
+        else:
+            raw_docs = Document.objects.raw(f'''SELECT * FROM doc_fts 
+                                                     WHERE text MATCH "{match.groups()[0]}";''')
             doc_ids = [doc.id for doc in raw_docs]
             doc_objs = doc_objs.filter(id__in=doc_ids)
 
-    except:
-        print("Error getting phrase")
+    if search_params['author'] != '':
+        author = search_params['author'].split(" ")
+        doc_objs = doc_objs.filter(Q(author_person__first__icontains=author[0]) |
+                                   Q(author_person__last__icontains=author[0]) |
+                                   Q(author_organization__name__icontains=author[0]))
 
-    try:
-        author = request.GET['author']
-        if author != "":
-            author = author.split(" ")
-            doc_objs = doc_objs.filter(Q(author_person__first__icontains=author[0]) |
-                                       Q(author_person__last__icontains=author[0]) |
-                                       Q(author_organization__name__icontains=author[0]))
-    except:
-        print("Error getting author name")
-    try:
-        recipient = request.GET['receiver']
-        if recipient != "":
-            recipient = recipient.split(" ")
-            doc_objs = doc_objs.filter(Q(recipient_person__first__icontains=recipient[0]) |
-                                       Q(recipient_person__last__icontains=recipient[0]) |
-                                       Q(recipient_organization__name__icontains=recipient[0]))
-    except:
-        print('Error getting recipient name')
-    try:
-        doc_types = request.GET['doc_type'].split(',')
-        print(doc_types)
-        if isinstance(doc_types, list) and 'unknown' not in doc_types:
-            queries = [Q(type__icontains=t) for t in doc_types]
-            print(queries)
-            query = queries.pop()
-            for item in queries:
-                query |= item
-        elif doc_types != 'unknown':
-            query = Q(type__icontains=doc_types)
-        doc_objs = doc_objs.filter(query)
-    except:
-        print('Error getting doc types')
-    try:
-        pages = [int(request.GET['minPages']), int(request.GET['maxPages'])]
-        doc_objs = doc_objs.filter(Q(number_of_pages__gte=pages[0]) &
-                                   Q(number_of_pages__lte=pages[1]))
-    except:
-        print('Error getting pages')
+    if search_params['recipient'] != '':
+        recipient = search_params['recipient'].split(" ")
```

Comment:
Same question here as for author.  Seems like what we want are queries like `Q(recipient_person__first__in=recipient`. Trouble is this is not case insensitive. and there's no __iin=. Hmm.